patroni/dcs/__init__.py

```diff
@@ -864,6 +1190,62 @@ def wrapper(*args: Any, **kwargs: Any):
 
 
 class AbstractDCS(abc.ABC):
+    """Abstract representation of DCS for preparing and loading :class:`Cluster` objects.
+
+    Implementations of a concrete DCS class, using appropriate backend client interfaces, must include the following
+    methods and properties.
+
+    Functional methods that perform construction of complex data objects:
+
+        * :meth:`~AbstractDCS._cluster_loader`:
+            method which processes the structure of data stored in the DCS used to build the :class:`Cluster` object
+            with all relevant associated data.
+        * :meth:`~AbstractDCS._citus_cluster_loader`:
+            Similar to above but specifically representing Citus group and workers information.
+        * :meth:`~AbstractDCS._load_cluster`:
+            main method for calling specific ``loader`` method to build the :class:`Cluster` object representing the
+            state and topology of the cluster.
```

Comment:
They all time critical. If didn't succeed during `retry_timeout` the primary is demoted because of lack of access to DCS.

Reply:
Thanks for these comments on this section. I'll bundle them together in one shuffle commit.